sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
             });
         });
     },
+    
+    onDemonstrationDelete: function(demonstration, operation) {
+        var me = this,
+            dashboardView = me.view,
+            mainGridEl = dashboardView.el.down('.cbl-grid-main'),
+            studentId = demonstration.get('StudentID'),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            skillStudentsTpl = dashboardView.getTpl('skillStudentsTpl'),
+
+            deletedSkills = demonstration.get('Skills'),
+            deletedSkillsLength = deletedSkills.length, deletedSkillIndex = 0, demonstratedSkill,
+            loadedCompetency, loadedSkill, demonstrationsByStudent, skillDemonstrationsCell, deletedSkillIds, studentDemonstrations, studentDemonstrationIndex = 0,
+
+            competencyCompletions = operation.getProxy().getReader().rawData.data[0].competencyCompletions,
+            competencyCompletionsLength = competencyCompletions.length, competencyCompletionIndex = 0, competencyCompletion,
+            competencyProgressCell, competencyPercent,
+            _compentencyDemonstrationResetFn,
+            _updateCompetencySkillsFn, 
+
+            updatesCompetencies = [],
+            updatesLength, updateIndex, update, skillsRows, skillHeadersRow, skillStudentsRow, competencySkills, skill,
+            
+            updatedDemonstrationLength, updatedDemonstrationIndex;
+
+
+        // compile operations and DOM references for needed updates -- don't write to dom!
```

Comment:
Batching DOM reads and writes has nothing to do with ExtJS or Javascript best practices, it's a DOM manipulation performance best practice